### PR TITLE
Mat 384 calculator improvements

### PR DESCRIPTION
**Pull request by bdsl**

### CODE REVIEW

File: src/Domain/CardBrand.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        if ($brand === null) {
+            return null;
+        }
+
+        return self::from($brand);
+    }
+
+    /**
+     * Whether this is amex or not is important Amex is more important to process.
```
Comment (bdsl) [inline src/Domain/CardBrand.php:35]:
```suggestion
     * Whether this is amex or not is important Amex is more expensive to process.
```

**Review by dorota-joanna — APPROVED** [review]
LGTM

**Comment by bdsl** [comment]
Thanks @dorota-joanna . I'll wait for @NoelLH to also take a look before merging.

File: tests/Domain/DonationServiceTest.php [hunk 2]
```diff
@@ -214,4 +216,57 @@ private function getDonation(): Donation
             TestCase::someCampaign(stripeAccountId: 'STRIPE-ACCOUNT-ID')
         );
     }
+
+    /**
+     * Not attempting to cover all possible variations here, just one example of confirming via the
+     * \MatchBot\Domain\DonationService::confirmOnSessionDonation . Details of the fee calcuations are
+     * tested directly against the Calculator class.
+     */
+    public function testItConfirmsOnSessionUKVisaDonationChargingApproiateFee(): void
```
Comment (NoelLH) [inline tests/Domain/DonationServiceTest.php:225]:
```suggestion
    public function testItConfirmsOnSessionUKVisaDonationChargingAppropriateFee(): void
```

**Review by NoelLH — APPROVED** [review]
Nice! LGTM

**Pull request merged**